qutip/tests/solver/test_mesolve.py

```diff
@@ -680,3 +680,19 @@ def test_mesolve_bad_state():
 def test_mesolve_bad_options():
     with pytest.raises(TypeError):
         MESolver(qutip.qeye(4), [], options=False)
+
+
+def test_feedback():
+
+    def f(t, A):
+        return (A-4.)
+
+    N = 10
+    tol = 1e-14
+    psi0 = qutip.basis(N, 7)
+    a = qutip.destroy(N)
+    H = qutip.QobjEvo(qutip.num(N))
+    solver = qutip.MESolver(H)
+    solver.add_feedback("A", qutip.num(N))
```

Comment:
```suggestion
    H = qutip.QobjEvo(qutip.num(N))
    solver = qutip.MESolver(H)
    solver.add_feedback("A", qutip.num(N))
```
I still really don't like the above pattern. `H` is can't sensibly be used by a solver without the specified feedback, so one should immediately be able to introspect `H` to find out what feedback it needs.

If we have that, the solver can look at `H` to see what feedback is needed and either supply that feedback or raise an error if it is unable to supply it.

This way mistakes cannot happen where `H` is silently not given the feedback it need and defaults to some initial value that is wrong, and we don't have `solver.add_feedback` repeatedly updating and overriding information on `H`.